CMakeLists.txt

```diff
@@ -0,0 +1,969 @@
+foreach ( mod ${SLI_MODULES} )
+  # get class name, is always in nest namespace
+  file( STRINGS "${PROJECT_SOURCE_DIR}/${mod}/${mod}module.h" module_class_string REGEX "class.*: public SLIModule" )
+  string( REGEX REPLACE "class ([a-zA-Z0-9_]+) : public SLIModule" "\\1" module_class ${module_class_string} )
+  file( APPEND ${PROJECT_BINARY_DIR}/nest/static_modules.h "  engine.addmodule(new nest::${module_class}());\n" )
+endforeach ()
+
+# when we build statically, we need to add headers and addmodule for external modules
+# just as if it were a in source module.
+if ( static-libraries AND external-modules )
+  file( APPEND ${PROJECT_BINARY_DIR}/nest/static_modules.h "\n  // Add all external modules:\n" )
+  foreach ( mod_header ${EXTERNAL_MODULE_INCLUDES} )
+    # get namespace:
+    file( STRINGS "${mod_header}" module_namespace_string REGEX "^namespace.*" )
+    if ( NOT module_namespace_string )
+      message( FATAL_ERROR "Could not find namespace in '${mod_header}'." )
+    endif ()
+    string( REGEX REPLACE "namespace ([a-zA-Z0-9_]+)" "\\1" module_namespace ${module_namespace_string} )
+
+    # get class name
+    file( STRINGS "${mod_header}" module_class_string REGEX "^class.*: public SLIModule" )
+    if ( NOT module_class_string )
+      message( FATAL_ERROR "Could not find class that extends SLIModule in '${mod_header}'." )
+    endif ()
+    string( REGEX REPLACE "class ([a-zA-Z0-9_]+) : public SLIModule" "\\1" module_class ${module_class_string} )
+    file( APPEND ${PROJECT_BINARY_DIR}/nest/static_modules.h "  engine.addmodule(new ${module_namespace}::${module_class}());\n" )
+  endforeach ()
+endif ()
+
+file( APPEND ${PROJECT_BINARY_DIR}/nest/static_modules.h "}\n\n#endif\n" )
+# write static_modules.h
+
+# Set tics per ms / step
+if ( tics_per_ms )
+  set( HAVE_TICS_PER_MS ON )
+endif ()
+
+if ( tics_per_step )
+  set( HAVE_TICS_PER_STEP ON )
+endif ()
+
+if ( with-ps-arrays )
+  set( PS_ARRAYS ON )
+endif ()
+################################################################################
+##################            Find Libraries here             ##################
+################################################################################
+
+# Only find libLTDL if we link dynamically
+set( HAVE_LIBLTDL OFF )
+if ( NOT static-libraries )
+  find_package( LTDL )
+  if ( LTDL_FOUND )
+    set( HAVE_LIBLTDL ON )
+
+    include_directories( ${LTDL_INCLUDE_DIRS} )
+    link_libraries( ${LTDL_LIBRARIES} )
+  endif ()
+endif ()
+
+# Find readline
+set( HAVE_READLINE OFF )
+if ( with-readline )
+  if ( NOT ${with-readline} STREQUAL "ON" )
+    # a path is set
+    set( READLINE_ROOT_DIR ${with-readline} )
+  endif ()
+
+  find_package( Readline )
+  if ( READLINE_FOUND )
+    set( HAVE_READLINE ON )
+    include_directories( ${READLINE_INCLUDE_DIRS} )
+    link_libraries( ${READLINE_LIBRARIES} )
+  endif ()
+endif ()
+
+# Find GSL
+set( HAVE_GSL OFF )
+set( HAVE_GSL_1_2 OFF )
+set( HAVE_GSL_1_11 OFF )
```

Comment:
I wonder if we should still differentiate between these GSL versions. V 1.11 was released in 2008.